README.md

```diff
@@ -37,6 +31,77 @@ Here is a short preview of it looks like currently:
 A roadmap is available in the issue [#5](https://github.com/prometheus-community/codemirror-promql/issues/5).
 
 ## Usage
+
+As the setup of the promQL language can a bit tricky in CMN, this lib provides a class `PromQLExtension` 
+which is here to help you to configure the different extension we aim to provide.
+
+### Default setup
+If you want to enjoy about the different features provides without taking too much time to understand how to configure it,
+then the easiest way is this one:
+
+```typescript
+import { PromQLExtension } from 'codemirror-promql';
+import { basicSetup, EditorState, EditorView } from '@codemirror/next/basic-setup';
+
+const promQL = new PromQLExtension()
+new EditorView({
+      state: EditorState.create({
+        extensions: [basicSetup, promQL.asExtension()],
+      }),
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      // tslint:disable-next-line:no-non-null-assertion
+      parent: document.getElementById('editor')!,
+    });
+```
+
+Using the default setup will activate:
+* the syntax highlight
+* an offline autocompletion that will suggest every promQL keyword such as the function / the aggregation depending of the context.
+* an offline linter that will display the promQL syntax error (which is closer to what Prometheus returns)
+
+### Deactivate autocompletion - linter
+In case you would like to deactivate, or/and the linter or the autocompletion it's simple as that:
+
+```typescript
+const promQL = new PromQLExtension().activateLinter(false).activateCompletion(false) // here the linter and the autocomplete are deactivated
+```
+
+### Connect the autocompletion extension to a remote Prometheus server
+Connecting the autocompletion extension to a remote Prometheus server will provide autocompletion of the metric/ the label name and the label value
+
+#### Use the default prometheus client
+
+##### Prometheus URL
+If you want to use the default prometheus client provided by this lib, you have to provide the url used to contact the Prometheus server.
+
+```typescript
+const promQL = new PromQLExtension().setComplete({url: 'https://prometheus.land'})
+```
+
+##### Override FetchFn
+In case your prometheus server is protected and required a special http client, you can override the function fetchFn that is used to perform any required HTTP request.
```

Comment:
```suggestion
In case your Prometheus server is protected and requires a special HTTP client, you can override the function `fetchFn` that is used to perform any required HTTP request.
```